### PR TITLE
Enable array input for KCL fields that need it

### LINKED ISSUE #8209: Add array validation support for KCL input fields
## Problem Description

Currently, array syntax like `[0.1, 0, 1]` is **not accepted** anywhere in KCL input fields throughout the application. However, some fields legitimately need to accept array values, specifically:

- Pattern Circular 3D: `axis` parameter (e.g., `[1, 0, 0]`)
- Pattern Circular 3D: `center` parameter (e.g., `[0, 0, 0]`)

When users try to enter array values in these fields, they get `'NAN'` errors instead of the arrays being properly parsed and formatted.

## Current Behavior

- All KCL input fields reject arrays and return `'NAN'` when array syntax is entered, because the `getCalculatedKclExpressionValue` function in `kclHelpers.ts` doesn't handle array formatting.

## Expected Behavior

**✅ Arrays should be accepted in fields that expect array values:**
- Pattern Circular 3D: `axis` parameter (e.g., `[1, 0, 0]`)
- Pattern Circular 3D: `center` parameter (e.g., `[0, 0, 0]`)
- Future array-type parameters in other operations

**❌ Arrays should continue to be rejected (return `'NAN'`) in fields that expect scalar values:**
- All length/distance fields
- All angle fields
- All radius fields
- All numeric count fields (instances, etc.)
- Variable name fields

## Proposed Solution

- **Add conditional array support:**  
  Modify `getCalculatedKclExpressionValue` and related functions to accept an optional `allowArrays` parameter (default: false).
- **Context-aware validation:**  
  Update `useCalculateKclExpression` hook to accept and pass through array validation context.
- **Smart field detection:**  
  Implement logic in `CommandBarKclInput` to automatically determine when arrays should be allowed based on command name and field name.
- **Explicit validation:**  
  Ensure other components continue to disallow arrays by default.

## Files to Modify

- `kclHelpers.ts` – Add array handling when `allowArrays: true`
- `useCalculateKclExpression.ts` – Add `allowArrays` parameter
- `CommandBarKclInput.tsx` – Auto-detect when arrays should be allowed
- `operations.ts` – Enable arrays for Pattern Circular 3D center parameter parsing

## Test Cases

- Arrays are accepted and properly formatted in Pattern Circular 3D axis field
- Arrays are accepted and properly formatted in Pattern Circular 3D center field
- Arrays continue to be rejected in all other KCL input fields
- Existing scalar value functionality continues to work unchanged
- Pattern Circular 3D operations work correctly with array inputs

## Implementation Notes

- This should be backward compatible – all existing behavior stays the same
- Arrays are disabled by default everywhere, enabled only for specific fields
- The change should be minimal and focused on just enabling arrays where needed
- Add unit tests for the new array formatting logic

### PR DESCRIPTION
**Pull request by max-mrgrsk**
Fixes #8209

### Problem
Users couldn't enter array syntax like `[1, 0, 0]` in Pattern Circular 3D `axis` and `center` parameter fields – got 'NAN' errors instead of proper array formatting.

### Solution
- Arrays now work for Pattern Circular 3D `axis` and `center` fields only
- All other KCL fields still reject arrays
- Only numeric arrays allowed—mixed types and empty arrays blocked

### Implementation
- `kclHelpers.ts`: `allowArrays` param, numeric validation
- `useCalculateKclExpression.ts`: threads `allowArrays`
- `CommandBarKclInput.tsx`: smart field detection

### Key Features
- Backward compatible
- Type safe (numbers, floats, units only)
- Arrays auto-enabled only where needed
- Strong validation

### Tests
10 tests: valid (`[1, 2, 3]`, `[1mm, 2mm, 3mm]`), invalid (`[1, true, 0]`, `[]`), edge cases

### Before / After
`[1, 0, 0]` → 'NAN'  
`[1, 0, 0]` → `[1, 0, 0]`

### CODE REVIEW

File: src/components/CommandBar/CommandBarKclInput.tsx [hunk 1]
```diff
@@ -120,6 +120,24 @@ function CommandBarKclInput({
...
+    // For Pattern Circular 3D operations, allow arrays for axis and center parameters
+    if (
+      commandName === 'patternCircular3d' &&
+      (argName === 'axis' || argName === 'center')
+    ) {
+      return true
+    }
+
+    // Add more cases here as needed for other operations that should accept arrays
+    // For now, disable arrays everywhere else (maintaining current production behavior)
```
Comment (pierremtb) [inline src/components/CommandBar/CommandBarKclInput.tsx]: Not a blocker, can be a follow up: it think having `allowArrays` as a property defaulting to false on kcl input types would be nicer to work with as we grow this?

File: src/lib/kclHelpers.ts [hunk 2]
```diff
@@ -15,7 +15,10 @@ const DUMMY_VARIABLE_NAME = '__result__'
  * Calculate the value of the KCL expression,
  * given the value and the variables that are available in memory.
  */
-export async function getCalculatedKclExpressionValue(value: string) {
+export async function getCalculatedKclExpressionValue(
+  value: string,
+  allowArrays: boolean = false // If true, allows numeric arrays only (e.g., [1, 2, 3])
```
Comment (jtran) [inline src/lib/kclHelpers.ts]: Instead of comments at the end of the line, prefer JSDoc style comments using [`@param`](https://jsdoc.app/tags-param) so that IDEs show them when you hover over the function.

File: src/lib/kclHelpers.ts [hunk 3]
```diff
@@ -38,6 +41,56 @@ export async function getCalculatedKclExpressionValue(value: string) {
...
+      return {
+        astNode: variableDeclaratorAstNode,
+        valueAsString,
+      }
+    }
+
+    // Validate that ALL array elements are numbers (required for geometric operations like patternCircular3d)
+    // This ensures arrays like [0, true, 0] are rejected, while [0, 1, 0] or [1mm, 2mm, 3mm] are accepted
+    const allElementsAreNumbers = varValue.value.every(
+      (item: any) => item.type === 'Number'
```
Comment (jtran) [inline src/lib/kclHelpers.ts]: Can we remove `any`? It's not safe.

File: src/components/CommandBar/CommandBarKclInput.tsx (hunk 1 above)
Comment (max-mrgrsk) [reply src/components/CommandBar/CommandBarKclInput.tsx]: [6f1fe3a - allowArrays as a property defaulting to false on kcl input types
](https://github.com/KittyCAD/modeling-app/pull/8275/commits/6f1fe3a84d62d611e7e6b01b3e45df3b0d3e76cf)

File: src/lib/kclHelpers.ts (hunk 2 above)
Comment (max-mrgrsk) [reply src/lib/kclHelpers.ts]: [5a44098 - param](https://github.com/KittyCAD/modeling-app/pull/8275/commits/5a440987af14d89514a6d911828a7105dabe8bf5)

File: src/lib/kclHelpers.ts (hunk 3 above)
Comment (max-mrgrsk) [reply src/lib/kclHelpers.ts]: [ed25856 - removed "any"](https://github.com/KittyCAD/modeling-app/pull/8275/commits/ed258562009be72f806c0138c78236b6dfbd2e2a)

Comment (max-mrgrsk) [reply src/lib/kclHelpers.ts]: @jtran what do you think about this fix?

File: src/components/CommandBar/CommandBarKclInput.tsx (hunk 1 above)
Comment (max-mrgrsk) [reply src/components/CommandBar/CommandBarKclInput.tsx]: @pierremtb what do you think?

File: src/lib/kclHelpers.ts (hunk 3 above)
Comment (lee-at-zoo-corp) [reply src/lib/kclHelpers.ts]: `& { type: 'Number' }` isn't correct. See [this example](https://www.typescriptlang.org/play/?#code/PTAEBUAsFNQewE4EsDmSB2BDANqADgnHnAM7QAmoAZkgB6gDuS2uDiA1qElaJupXHTYAnlx4ADAC7C80cVxL4E0Muknx0vFqABuOAK4qAdAChps0AGkAxtgBqB2AF5QAb1DnoALlAByEpLI6Ci+ADSg5JiSmD4BQSigAL6gAD5uHjLefuj6ALYARtAIYRFRMaA5BUVJqekZsj6++diYkL5JZplWtg7YhgByeYUIoC429o6gAGTpno2Vw+2JJtjQ6uXjvQND1S7uc9k7xeGR0T4AzB0mIKAAquhUiJL6WJLQIlzqTNqeALTWMGsnBeGDeaiQghwHzgkhgCCYZC8nQsm0cACZRrNMo04hgQicyrFAniaml9tjDlVjqUzhUjqS6nB2D4FtVlp5uhM+tBBlSMWMeujpliGpTFh0bgB1WAMPjreqwKiYZj6ZQAQjVKzWoHyPlR3N5w35IqyvlZ1KZPgALFcblBYNZEMprF9MKJJHAdbBkChEHB9IpMPl-eokOpCiSAIzwHiw0iwEGCRRIRSO-hhiFYXCeEhIjn6wyXPYKnHE4IlU7lXHBBnk0Vmo4Vwl0qm10DwZkt4YdfOCg1Hc4AHnAmIL0HOAD5MSPoLQweRFHXTealqAAPwQUAs6A6IoAbmuYAAQnA4MJMOEOUrmCRNat1NY9X3tlSiyb5o3wpbQDbEkA) (Edit: I've had to update the playground code a few times, please check from GitHub and not an email)

Comment (jtran) [reply src/lib/kclHelpers.ts]: Yeah, I like Lee's solution if you can get that to work. Ask one of us for help if you need it.

Comment (max-mrgrsk) [reply src/lib/kclHelpers.ts]: @lee-at-zoo-corp [773a409](https://github.com/KittyCAD/modeling-app/pull/8275/commits/773a4096ea7389612804c1fcf56197f7f985b969)

File: src/lib/kclHelpers.ts [hunk 4]
```diff
@@ -38,6 +51,52 @@ export async function getCalculatedKclExpressionValue(value: string) {
     resultDeclaration?.type === 'VariableDeclaration' &&
     resultDeclaration?.declaration.init
   const varValue = execState.variables[DUMMY_VARIABLE_NAME]
+
+  // Handle array values when allowArrays is true
+  if (
+    allowArrays === true &&
```
Comment (jtran) [inline src/lib/kclHelpers.ts]:
```suggestion
    allowArrays &&
```

File: src/lib/kclHelpers.ts (hunk 3 above)
Comment (lee-at-zoo-corp) [reply src/lib/kclHelpers.ts]: The Leent is satisfied
